File: routes/folderRoutes.js

const express = require('express');
const { createFolder, getUserFolders, updateFolder, deleteFolder, getFolderById, getFoldersByType } = require('../controllers/folderController');
const router = express.Router();

// Create a new folder
router.post('/', createFolder);

// Get all folders for the logged-in user
router.get('/', getUserFolders);
router.get('/type/:type', getFoldersByType);

// Get a specific folder by ID
router.get('/:id', getFolderById);

// Update a folder
router.put('/:id', updateFolder);

// Delete a folder
router.delete('/:id', deleteFolder);

module.exports = router;

File: routes/folder_routes.py
from fastapi import APIRouter
from controllers.folder_controller import FolderController

router = APIRouter(prefix="/folders", tags=["Folders"])

@router.post("/", response_model=dict)
async def create_folder(data: dict):
    return await FolderController.create_folder(data)

@router.get("/", response_model=list)
async def get_user_folders():
    return await FolderController.get_user_folders()

@router.get("/type/{type}", response_model=list)
async def get_folders_by_type(type: str):
    return await FolderController.get_folders_by_type(type)

@router.put("/{id}", response_model=dict)
async def update_folder(id: str, data: dict):
    return await FolderController.update_folder(id, data)

@router.delete("/{id}", response_model=dict)
async def delete_folder(id: str):
    return await FolderController.delete_folder(id)

@router.get("/search", response_model=list)
async def search_folders(query: str):
    return await FolderController.search_folders(query)